mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -131,7 +131,7 @@ class RowMatrix(
       throw new IllegalArgumentException(s"Argument with more than 65535 cols: $cols")
     }
     if (cols > 10000) {
-      val mem = cols * cols * 8
+      val mem = (math.pow(cols, 2) * 8).formatted("%,.0f")
```

Comment:
I know it's minor, but I think it might be simpler to report megabytes in the warning message, since the reported size is at least 800MB. Then this can be simply `val memMB = (cols.toLong * cols) / 125000`; that multiplication can't overflow.